crates/biome_js_analyze/src/lint/nursery/no_useless_undefined_initialization.rs

```diff
@@ -0,0 +1,135 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_syntax::JsVariableDeclaration;
+use biome_rowan::{AstNode, BatchMutationExt, TextRange};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Disallow initializing variables to `undefined`.
+    ///
+    /// A variable that is declared and not initialized to any value automatically gets the value of `undefined`.
+    /// It’s considered a best practice to avoid initializing variables to `undefined`.
+    ///
+    /// Please note that any inline comments attached to the initialization value or variable will be removed on auto-fix.
+    /// Please be also aware that this differs from Eslint's behaviour and we are still discussing on how to properly handle this case.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = undefined;
+    /// ```
+    /// ```js,expect_diagnostic
+    /// let b = undefined, c = 1, d = undefined;
+    /// ```
+    /// ```js,expect_diagnostic
+    /// for (let i = 0; i < 100; i++) {
+    /// 	let i = undefined;
+    /// }
+    /// ```
+    /// ```js,expect_diagnostic
+    /// let f = /**/undefined/**/ ;
+    /// ```
+    /// ### Valid
+    ///
+    /// ```js
+    /// var a = 1;
+    /// ```
+    /// ```js
+    /// class Foo {
+    /// 	bar = undefined;
+    /// }
+    /// ```
+    ///
+    pub NoUselessUndefinedInitialization {
+        version: "next",
+        name: "noUselessUndefinedInitialization",
+        sources: &[RuleSource::Eslint("no-undef-init")],
```

Comment:
Since we decided to diverge, you need to add a new metadata called `source_kind` and mark it as `Inspired`. Look how other rules do it